powersimdata/network/model.py

```diff
@@ -14,9 +15,15 @@ class ModelImmutables:
 
     :param str model: grid model name.
     :param str interconnect: interconnect of grid model.
+    :param pandas.DataFrame zone: a data frame with loadzone name (*'zone_name'*),
+        division name (e.g. *'state'* name for USA grid models), interconnect name
+        (*'interconnect'*), time zone of loadzone (*'time_zone'*) and division
+        abbreviation (*'abv'*) as columns; and loadzone id (*'zone_id'*) as indices. If
+        None, it will be assumed that the data frame can be built from a CSV file
+        stored on disk.
```

Comment:
Wondering what if a `zone` data frame is provided but not complete

Reply:
I can create a function that checks that the shape is as expected and that the columns and index are correctly labelled.

Reply:
Done in last commit